torch/csrc/jit/passes/quantization.cpp

```diff
@@ -2709,6 +2716,66 @@ void ReplicateDeQuant(std::shared_ptr<Graph>& graph) {
   }
 }
 
+void ReplicateChooseQParamsQuant(std::shared_ptr<Graph>& graph) {
+  std::stack<Block*> blocks_to_visit;
+  std::vector<Node*> quant_nodes_to_rewrite;
+  std::vector<Node*> choose_qparam_nodes_to_rewrite;
+  blocks_to_visit.push(graph->block());
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if ((n->kind() == Symbol::aten("quantize_per_tensor") ||
+           n->kind() == Symbol::aten("quantize_per_channel")) &&
+          n->output()->uses().size() > 1) {
+        quant_nodes_to_rewrite.push_back(n);
+      }
+      for (Block* subblock : n->blocks()) {
+        blocks_to_visit.push(subblock);
+      }
+    }
+  }
+  for (Node* n : quant_nodes_to_rewrite) {
+    Value* original_val = n->inputs()[0];
+    Value* quantized_val = n->output();
+    Value* dtype = n->inputs()[3];
+    TORCH_CHECK(original_val->uses().size() == 2);
```

Comment:
what is this check?

Reply:
The original value should only have 2 uses i.e. choose_qparams and quantize_per_tensor. I